Tasks/Maven/maventask.ts

```diff
@@ -32,6 +32,35 @@ var reportPOMFile: string = null;
 var execFileJacoco: string = null;
 var ccReportTask: string = null;
 
+var isWindows = os.type().match(/^Win/);
+
+function readJavaHomeFromRegistry(jdkVersion: string, arch: string): string {
```

Comment:
in a future revision (in master) you may want to move this into a common internal lib.